/**********************************************/
Ganancias de aplicacion en AppStore y PlayStore
/**********************************************/

El objetivo de este proyecto es encontrar las ganancias de las palicaciones desplegadas en appstore y playstore

- en Septiembre del 2018 había aproximadamente 2 millones de apps disponobles el appStore y 2.1 millones de aplicaciones disponibles en PlayStore

- se tienen 2 data sets obtenidos de Kaggle.com

- Empecemos por abrir los 2 data sets

In [177]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv',encoding="utf-8")
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv', encoding='utf-8')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

para hacer la exploracion más sencilla los data sets, se generara funcion axplore_data() para usarla de forma repetitiva para explorara datos, se muestra numero de filas y columnas para cualquier set de datos.

In [178]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Numero de lineas:', len(dataset))
        print('Numero de columnas:', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Numero de lineas: 10841
Numero de columnas: 13


verificamos que el data set de Google contiene 10841 apps y 13 columnas, las columnas que pudieran sernos de utilidad son: 'App', 'Category', 'Reviews', 'Installs', 'Type', 'Price', and 'Genres'.

In [179]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Numero de lineas: 7197
Numero de columnas: 17


verificamos el dataset de iOS que contiene 7197 apps y 16 columnas

/***borrar datos erroneos***/, hoy un error en el registro 10472

In [180]:
print(android[10472])  # linea incorrecto
print('\n')
print(android_header)  # header
print('\n')
print(android[0])      # linea correcta

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


la lineas 10472 corresponde a la app Life Made WI-Fi Touchscreen Photo Frame, y tiene de rating '19', esto es incorrecto por q la calificacion maxima de Google Play es 5, asi que borramos esta linea.

In [181]:
print(len(android))
del android[10472]  # no correr esta lina mas de 1 vez
print(len(android))

10841
10840


/**** REMOVER DATOS DUPLICADOS ****/

PARTE UNO

Si exploramos el data set de Google podemos ver apps con más de 1 registro, por ejemplo Instagram

In [182]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


En total hay 1181 casos donde las apps aparecen más de una vez

In [183]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Numero duplicado de apps: ', len(duplicate_apps))
print('\n')
print('Ejemplo de apps duplicadas: ', duplicate_apps[:15])

Numero duplicado de apps:  1181


Ejemplo de apps duplicadas:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


No queremos contar ciertas applicaciones más de una vez al momento de analizar el dataset, así que eliminamos datos repetidos.

Si se examinan las lineas impresas, se observa una diferencia en la cuarta posicion de cada fila, la cual corresponde a al numero de revisiones. Los diferentes numeros muestran que los datos fueron recolectados en diferentes tiempos. Podemos usar estos datos para construir un criterio para las siguientes fila. No queremos removerlas de forma aleatoria, si no que nos quedamos con los numeros mas altos de revisiones.

Para hacer eso haremos lo siguiente:

* Crear un diccionario donde cada llave es el nombre unico de una app, y el como valor, la revision mas alta.
* Usar el diccionario para crea5r un nuevo data set, el cual tendra solo una entrada por app(y solo seleccionaremos la apps con el numero más alto de revisiones

Empecemos por construir el diccionario:

In [184]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
     

en la celda anterior encontramos 1181 casos donde una app aparece más de 1 vez, asi que la longitud de nuestro diccionario debe ser igual a la diferencia entre la longitud de nuestro data set y 1181


In [185]:
print('Longitud esperada', len(android) - 1181)
print('longitud actual' , len(reviews_max))

Longitud esperada 9659
longitud actual 9659


ahora usaremos el diccionario para remover dupliacdos, haremos lo siguiente:

* inicializamos 2 listas vacias *android_clean* y *already_added*
* recorremos el data set android y por cada iteracion haremos:
    * obtendremos el nombre de  la app y numero de revisiones
    * agregamos la linea a la lista *android_clean* y el nombre a la lista *already_cleaned* si:
        * el numero de revisiones de la app es igual a el número de revisiones de esa app descritra en el diccionario                     *reviews_max*, y:
        * el nombre de la app no se encuentra en la lista *already_added*. Necesitamos agregar esta condicion suplemenrtaria.

In [186]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name) #Esto debe ir dentro del bloque if

ahora exploremos rapidamente el nuevo data set y confirmamos que el numero de lineas son 9659

In [187]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Numero de lineas: 9659
Numero de columnas: 13


se obtienen 9659 lineas como era esperado

/**REMOVIENDO APLICACIONES QUE NO ESTAN EN INGLES**/

PARTE UNO

si exploramos el data set, notamos que hay nombres de aplicaciones en caracteres orientales, a continuacion se muestran algunos ejemplos:

In [188]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

436672029
1144164707
中国語 AQリスニング
لعبة تقدر تربح DZ


No nos interesa mantener estas apps en nuestro Data Set, asi que las removeremos. Un modo de hacer esto es eliminar cada nombre que contenga simbolos fuera de los caracteres especiales y otros simbolos (., !, ?, ;, etc.), (+, *, /, etc.).

- Construiremos una funcion y usaremos la funcion ord() para encontrar el encoding especifico a cada caracter.

In [189]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


Esta funcion parece funcionar de forma correcta, pero algunas aplicaciones usan en sus nombre emojis u otros simbolos  (™, — (em dash), – (en dash), etc.), eso cae fuera del rango ASCII, eso nos haria remover aplicsaciones que nos son utiles

In [190]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


*PARTE DOS*

Para minimizar el impacto de la perdida de datos solo removeremos las apps que en su nombre contengan mas de 3 caracteres *no-ASCII*

In [191]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii >= 1:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

False
False


bnuestra funcion no es perfecta, pudiera haber aplicaciones que pasan nuestro filtro, pero a este punto no debemos gastar mucho tiempo optimizando este filtro.

A continuacion usamos nuestra funcion *is_english()* para filtrar ambos data sets.

In [192]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Numero de lineas: 9117
Numero de columnas: 13


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2

verificamos resultados!!

*AISLANDO APLICACIONES GRATUITAS*

Como mencionamos al principio solo necesitamos para nuestro analisis las aplicaciones gratuitas; nuestros data sets contiene aplicaciones gratuitas y de paga, y necesitamos aislar las palicaciones gratuitas en ambos data sets.

In [193]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[5]
    if price == '0':
        ios_final.append(app)

        
print(len(android_final))
print(len(ios_final))

8408
4056


obtenemos 8864 apps Android y 4056 apps iOS, los cuales son suficientes para nuestro analisis.

*APLICACIONES MAS COMUNES POR GENERO*

-PARTE UNO-

* Como se menciono en la introduccion, nuestra meta es determinar el tipo de aplicaciones que son más atractivas a los usuarios, debido a que nuestros resultados son altamente influenciados por el número de personas que usan las aplicaciones.

para minimizar los riesgos, nuestra estrategia de validacion esta compuesta por 3 pasos:

1.- construir una version minima de la aplicacion Android y agregarla a Google Play.
2.- si la app tiene buena respuesta se desarrollara lo siguiente.
3.- si la aplicacion tiene ganancias despues de 6 meses, tambien se construira una Version para iOS de la aplicación.

-PARTE DOS-

construiremos 2 funciones que usaremos para analizar la frecuencia en las tablas

In [194]:
def freq_table(dataset, index):
    table= {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ":", entry[0])

*PARTE TRES*

- Empezamos a examinar la frencuencia en las tablas para la columna prime_genre del data set de App Store.

In [195]:
display_table(ios_final, -5)

Games : 55.64595660749507
Entertainment : 8.234714003944774
Photo & Video : 4.117357001972387
Social Networking : 3.5256410256410255
Education : 3.2544378698224854
Shopping : 2.983234714003945
Utilities : 2.687376725838264
Lifestyle : 2.3175542406311638
Finance : 2.0710059171597637
Sports : 1.947731755424063
Health & Fitness : 1.8737672583826428
Music : 1.6518737672583828
Book : 1.6272189349112427
Productivity : 1.5285996055226825
News : 1.4299802761341223
Travel : 1.3806706114398422
Food & Drink : 1.0601577909270217
Weather : 0.7642998027613412
Reference : 0.4930966469428008
Navigation : 0.4930966469428008
Business : 0.4930966469428008
Catalogs : 0.22189349112426035
Medical : 0.19723865877712032


podemos ver que entre las aplicaciones gratuitas mas de la mitad (55.15%) son juegos, entretenimiento(8.23%) seguido de aplicaciones de foto y video. Solo el 3.25% estan dedicadas a la educacion.

La impresion general es que la App Store(al menos la parte que contiene las aplicaciones gratuitas) esta dominada por aplicaciones diseñadas para el entretenimiento (juegos, entretenimiento, fotos y video), mientras que aplicaciones de proposito practico(educacion, compras, utilidades, productividad, estilo de vida, etc.) son mas escasas. Sin embargo, el hecho de que las aplicaciones para entretenimiento son mas numerosas no implica que tengan el mayor nuemro de usuarios, la demanda tal vez no sea igual que la oferta.

Continuemos examinando Genero y categoria del data set de Google Play

In [196]:
display_table(android_final, 1) # Category

FAMILY : 18.803520456707897
GAME : 9.60989533777355
TOOLS : 8.575166508087536
BUSINESS : 4.709800190294957
PRODUCTIVITY : 3.9724072312083734
LIFESTYLE : 3.8891531874405327
FINANCE : 3.73453853472883
MEDICAL : 3.6393910561370126
PERSONALIZATION : 3.306374881065652
SPORTS : 3.258801141769743
COMMUNICATION : 3.2231208372978117
HEALTH_AND_FITNESS : 3.1279733587059946
PHOTOGRAPHY : 3.0090390104662226
NEWS_AND_MAGAZINES : 2.7949571836346334
SOCIAL : 2.664129400570885
TRAVEL_AND_LOCAL : 2.3073263558515698
SHOPPING : 2.247859181731684
BOOKS_AND_REFERENCE : 2.1883920076117986
DATING : 1.8315889628924835
VIDEO_PLAYERS : 1.7602283539486203
MAPS_AND_NAVIGATION : 1.3558515699333968
FOOD_AND_DRINK : 1.2012369172216937
EDUCATION : 1.165556612749762
ENTERTAINMENT : 0.939581351094196
AUTO_AND_VEHICLES : 0.939581351094196
LIBRARIES_AND_DEMO : 0.9039010466222646
HOUSE_AND_HOME : 0.8087535680304472
WEATHER : 0.7968601332064701
EVENTS : 0.7136060894386299
ART_AND_DESIGN : 0.6660323501427212
PARENTING : 0.6

La principal diferencia que podemos ver es que no hay tantas apps destinadas al entretenimiento, y parece que un buen numero de apps esta diseñada a propositos practicos(familia, herramientas. estilo de vida, productividad,etc.). sin embargo si investigamos más podemos ver que la categoria familia(que tiene al rededor del 19%) son mayormente juegos para niños.

Verifiquemos ahora el genero de las apps en el dataset.

In [197]:
display_table(android_final, -4) #Genero

Tools : 8.563273073263558
Entertainment : 6.089438629876309
Education : 5.387725975261656
Business : 4.709800190294957
Productivity : 3.9724072312083734
Lifestyle : 3.8772597526165553
Finance : 3.73453853472883
Medical : 3.6393910561370126
Sports : 3.3301617507136063
Personalization : 3.306374881065652
Communication : 3.2231208372978117
Health & Fitness : 3.1279733587059946
Action : 3.116079923882017
Photography : 3.0090390104662226
News & Magazines : 2.7949571836346334
Social : 2.664129400570885
Travel & Local : 2.3073263558515698
Shopping : 2.247859181731684
Books & Reference : 2.1883920076117986
Simulation : 2.0813510941960036
Dating : 1.8315889628924835
Arcade : 1.8315889628924835
Casual : 1.7721217887725977
Video Players & Editors : 1.736441484300666
Maps & Navigation : 1.3558515699333968
Food & Drink : 1.2012369172216937
Puzzle : 1.1298763082778307
Racing : 1.0228353948620361
Role Playing : 0.939581351094196
Auto & Vehicles : 0.939581351094196
Strategy : 0.9039010466222646
Librar

la diferencia entre las columnas Genero y Categoria no es muy clara, pero podemos notar que la columna Genero es un poco más granular(tiene más categorias). a partir de ahora solo usaremos la columna Categoria.

hasta este punto encontramos que la App Store esta dominada por aplicaciones dideñadas para el entretenimiento, mientras que Google Play muestra más un mejor balance.

    *APLICACIONES MAS POPULARES POR GENERO EN LA APP STORE*
    
Un modo de de saber que Generos son los más populares(tienen mayor numero de usuarios) is calcular el promedio de instalaciones de aplicaciones de cada genero. Para el data set de Google Play podemos encontrar esta informacion en la columna Installs, pero para el data set de App Store esta informacion no se encuentra. Tomaremos como el total de numero de usuarios que rankearon cada app, esta info esta en la columna rating_count_tot.

Ahora calculamos el promedio de raitings por genero en el App Store.

In [198]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[6])
            total += n_ratings
            len_genre += 1

    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Productivity : 19053.887096774193
Weather : 47220.93548387097
Shopping : 18746.677685950413
Reference : 67447.9
Finance : 13522.261904761905
Music : 56482.02985074627
Utilities : 14010.100917431193
Travel : 20216.01785714286
Social Networking : 53078.195804195806
Sports : 20128.974683544304
Health & Fitness : 19952.315789473683
Games : 18924.68896765618
Food & Drink : 20179.093023255813
News : 15892.724137931034
Book : 8498.333333333334
Photo & Video : 27249.892215568863
Entertainment : 10822.961077844311
Business : 6367.8
Lifestyle : 8978.308510638299
Education : 6266.333333333333
Navigation : 25972.05
Medical : 459.75
Catalogs : 1779.5555555555557


en promedio las aplicaciones para navegar tiene el numero mas alto de revisionespero este numero esta fuertemente influenciado por WASE y Google Maps que tienen cerca de la mitad de las revisiones juntas:

In [199]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[2], ':', app[6]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
恵方コンパス. : 0
ナビタイム ドライブサポーター - NAVITIMEのカーナビアプリ : 0
自転車ナビ by NAVITIME(ナビタイム) - 自転車のナビができるアプリ : 0
百度地图-智能的手机导航，公交地铁出行必备 : 1014
高德地图（精准专业的手机地图） : 1040
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
ナビタイムの乗り換え案内 - 遅延情報やバス時刻表を案内するアプリ : 48
百度地图HD : 771
高德地图HD : 26
Google Maps - Navigation & Transit : 154911
JR東日本アプリ : 0
NAVIRO(ナビロー) - カーナビ/バイクナビ/徒歩ナビが使える高性能ナビアプリ : 0
ホラースポット-ghost spot-意味が分かると怖いマップ : 0
バーチャル恵方巻【節分・恵方コンパス・方位】 : 0
えほう - 最強の恵方コンパス : 0
MapFan(マップファン) – 渋滞情報/オービス/オフライン対応の本格カーナビ : 0


el mismo patron aplica para aplicaciones de redes socialesdonde la mayoria son descargas de facebook, Pinterest, Skype, etc. Lo mismo aplica para aplicaciones de musica, donde aplicaciones como Pandora, Spotify y Shazam influyen en el promedio de estas aplicaciones.

nuerstra meta es encontrar los generos mas populares pero, navegacion, redes sociales o musica parecen más populares de lo que realmente son. El promedio de ratings parece ser influenciado por muy pocas aplicaciones, mientras que otras tienen muy pocas descargas. Podemos generar una mejor imagen removiendo esas aplicaciones que son muy populares para cada genero y promediar nuevamente.

son al rededor de 74942 aplicaciones de referencia.

In [201]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[2], ':', app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
彩库宝典-【官方版】 : 0
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8
無料で音楽や写真・カメラの裏技アプリ for iPhone7 : 0


Con el analisis de estos generos podemos determinar varias cuestiones, parece que  las palicaciones más populares incluyen clima, libros, comida o finanzas; las aplicaciones que no parecen muy interesantes para nosotros son:

* Clima - la gente generalmente no pasa mucho tiempo en estas apps asi que las oportunidades de monetizar una aplicacion de este tipo son bajas.
* Camida y bebida - ejemplos de estaas apps son Starbucks, Dunkin' Donuts, McDonalds, etc. Asi que para generar una aplicacion popular de este tipo requiere preparacion y entrega, lo cual esta fuera de los limites de la empresa.
* Finanzas - estas aplicaciones involucran bancos, pagos, transferencias, etc. Construir una aplicacion financiera requiere de conocimiento y no queremos contratar a un financiero solo para contruir una app.

AHORA ANALICEMOS EL DATA SET DE GOOGLE PLAY

*APLICACIONES MAS POPULARES POR GENERO EN EL GOOGLE PLAY*

Para el Google Play tenemos el dato de numero de instalaciones, asi que debemos de ser capaces de obtener una imagen más clara acerca de la popularidad de las palicaciones. Sin embargo el numero de instalaciones no parece muy acertado

In [202]:
display_table(android_final, 5) # columna de numero de instalaciones

1,000,000+ : 15.592293054234062
100,000+ : 11.596098953377735
10,000+ : 10.442435775451951
10,000,000+ : 10.323501427212179
1,000+ : 8.480019029495718
100+ : 7.088487155090391
5,000,000+ : 6.660323501427213
500,000+ : 5.5542340627973354
50,000+ : 4.7216936251189345
5,000+ : 4.5313986679353
10+ : 3.5442435775451955
500+ : 3.246907706945766
50,000,000+ : 2.2121788772597526
100,000,000+ : 2.1289248334919124
50+ : 1.9743101807802093
5+ : 0.8206470028544244
1+ : 0.5114176974310181
500,000,000+ : 0.285442435775452
1,000,000,000+ : 0.22597526165556614
0+ : 0.04757373929590866
0 : 0.011893434823977166


un problema con los datos es que no son precisos. Por ejemplo no sabemos si una app con 100,000+ instalaciones tiene 200,000 o 350K, sin embargo no necesitamos ser muy precisos para nuestro proposito, solo queremos una idea de que genero de aplicacion es más atractiva, y no necesitamos una precision perfecta.

en el siguiente proceso vamos a convertir cada numero de instalacion en *float* - eso significa que debemos quitar las comas y los signos de adicion, de otro modo la conversion no se podra producir - lo haremos en el siguiente ciclo donde tambien obtendremos promedio de instalacion de cada genero(categoria)

In [ ]:
categorias_android = freq_table(android_final, 1)

for category in categorias_android:
    total = 0
    len_categoria = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            